Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStore.cs

```diff
@@ -358,11 +366,27 @@ protected override async Task<BoolResult> ShutdownCoreAsync(OperationContext con
         }
 
         /// <nodoc />
-        internal Task<BoolResult> HeartbeatAsync(OperationContext context)
+        internal async Task<BoolResult> HeartbeatAsync(OperationContext context)
         {
-            return context.PerformOperationAsync(
-                Tracer,
-                () => ProcessStateAsync(context, inline: false));
+            // This makes sure that the heartbeat is only run once for each call to this function. It is non-blocking
+            // wait, so no need to use the async version.
+#pragma warning disable AsyncFixer02
+            if (!_heartbeatGate.Wait(0))

```

Comment:
One way to remove these pragmas in two places and simplify the code a bit is to change WaitToken extension method from SemaphoreSlimExtension to take the wait time or to add an extension method called TryWait() that will return SemaphoreSlimToken (a disposable instance).

In this case you'll remove two pragmas and will be able to use this:

bool acquired
using _gate.TryWait(out var acquierd);
if (!acquired)
{
  return BoolResult.Success;
}

return await context.PerformOperationAsync(

What do you think?